src/st/lib/e_lesser_demon.c

```diff
@@ -1,8 +1,59 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 #include "lib.h"
 
-// Lesser demon start
-INCLUDE_ASM("st/lib/nonmatchings/e_lesser_demon", func_us_801BB8DC);
+// Seems to be related to the iframes of the ectoplasm and the skeleton spawns
+// Mudman spawn animation causes iframes for it
+void func_us_801BB8DC(s16* unkArg) {
+    switch (g_CurrentEntity->step_s) {
+    case 0:
+        g_CurrentEntity->animCurFrame = 0;
+        g_CurrentEntity->hitboxState = 0;
+        g_CurrentEntity->zPriority -= 0x10;
+        g_CurrentEntity->ext.lesserDemon.unkB2 = g_CurrentEntity->palette;
+        g_CurrentEntity->drawFlags |= FLAG_DRAW_UNK8;
+        g_CurrentEntity->unk6C = 2;
+        g_CurrentEntity->step_s++;
+        g_CurrentEntity->flags |= FLAG_UNK_2000;
+        g_CurrentEntity->flags &= ~(FLAG_UNK_800 | FLAG_UNK_400);
+        if (!unkArg) {
+            g_CurrentEntity->step_s = 2;
+        }
+        break;
+    case 1:
+        if (UnkCollisionFunc3(unkArg) & 1) {
+            g_CurrentEntity->animCurFrame = 1;
+            g_CurrentEntity->step_s++;
+        }
+        break;
+    case 2:
+        g_CurrentEntity->unk6C += 4;
+        if (g_CurrentEntity->unk6C > 0xA0) {
+            g_CurrentEntity->drawFlags = FLAG_DRAW_DEFAULT;
+            g_CurrentEntity->drawMode = DRAW_DEFAULT;
+            g_CurrentEntity->ext.lesserDemon.unkB0 = 0x20;
+            g_CurrentEntity->step_s++;
+        }
+        break;
+    case 3:
+        if (g_CurrentEntity->ext.lesserDemon.unkB0 % 2) {
+            g_CurrentEntity->palette = g_CurrentEntity->ext.lesserDemon.unkB2;
+        } else {
+            g_CurrentEntity->palette = 0x819F;
+        }
+
+#ifdef VERSION_PSP
+        if (!(--g_CurrentEntity->ext.lesserDemon.unkB0)) {
+#else
+        if (!(--g_CurrentEntity->ext.lesserDemon.unkB0 << 16)) {
```

Comment:
Don't think this is necessary? `if (!(--g_CurrentEntity->ext.et_lesserDemon.unkB0))` seems to match both PSX and PSP